doc/developer-notes.md

```diff
@@ -771,6 +771,8 @@ Wallet
 General C++
 -------------
 
+As of v27.0, we require a compiler which meets at least the C++20 standard.
```

Comment:
It's useful to document what version of the C++ standard can be used, but i don't think this needs to mention the version it was introduced (the document always applies to the branch it is in), nor is this really the place for documenting a compiler requirement.

Would word it as "Code needs to adhere to the C++20 standard.".

Reply:
> Would word it as "Code needs to adhere to the C++20 standard.".

I second this.

Reply:
Sure, wasn't sure this was the place, but is there somewhere better?  Seems like all the regulars "know" this, so putting it where a beginner might look is helpful.

Was thinking the version info is useful for backporting, but that's a limited window and is already in the commit message if people want.

Reply:
I think adding this info here as you propose with the wording suggested by @laanwj would have some chance to be merged (I would ack it). Just don't expect it to attract much attention of reviewers since lacking this info doesn't seem to create much problem.

About backporting, even small nits of documentation are somewhat complex and is reserved to be done by a few maintainers from time to time. I don't believe it will happen with this since this is not critical and doesn't apply to earlier versions.

Reply:
Ah, I see what you're saying.  I was prefer "you can assume c++20" (which was my issue, using <bit>, not "you must meet c++20" which implies you understand all nuances of C++ standards?

Reply:
Sure, i'm fine with wording it in a way that doesn't imply people understand all nuances, say "The code can use the C++20 standard.", but due to the context it should be a statement about the code, not a compiler requirement.

Reply:
> "you can assume c++20"

Note that it depends on [compiler support](https://en.cppreference.com/w/cpp/compiler_support/20) for the individual features; for instance [modules](https://en.cppreference.com/w/cpp/language/modules) are nominally part of C++20, but still have pretty poor compiler support, so aren't something we can make use of. Conversely, sometimes if a feature is widely supported by compilers/libs, we'll use it even if it's not part of the current language standard (eg #24531 #25472).

In that context, not sure if documenting C++20 is all that helpful; if you're unsure, seems better to ask ("Discuss project-specific development on #bitcoin-core-dev on Libera Chat" per doc/README.md). No particular objection either though.

> which was my issue, using <bit>, 

For `<bit>` in particular, that feature's been pulled in by #29085 a few months ago. The C++20 PR was #28349 and discussion was in #23363 since they don't seem to have already been mentioned here.

Reply:
Right, in practice not even all C++11 features are fully supported, nor always desirable, e.g. don't `thread_local` unless it's for a very good reason and only for POD types. But as starting point before listing all the exceptions i think it still makes sense to mention.